esp/src/src-react/util/metricGraph.ts

```diff
@@ -193,18 +241,21 @@ subgraph cluster_${sg.id} {
         this._dedupEdges = {};
         const childTpls: string[] = [];
         if (items?.length) {
-            items.forEach(item => {
+            items.map(item => {
                 if (this.subgraphExists(item.id)) {
-                    childTpls.push(this.subgraphTpl(item, options));
+                    return item;
                 } else {
-                    if (item?.__parentID && this.subgraphExists(item?.__parentID)) {
-                        childTpls.push(this.subgraphTpl(this.subgraph(item.__parentID), options));
+                    if (item?.__parentName && this.subgraphExists(item?.__parentName)) {
+                        return this.subgraph(item.__parentName);
                     }
                 }
+            }).filter(subgraph => !!subgraph).map(subgraph => {
```

Comment:
I can't remember if I've used this shorthand filter like this before, is it removing undefined items?